advanced/index.php

```diff
@@ -1,81 +0,0 @@
-<?php
-/* Pi-hole: A black hole for Internet advertisements
-*  (c) 2017 Pi-hole, LLC (https://pi-hole.net)
-*  Network-wide ad blocking via your own hardware.
-*
-*  This file is copyright under the latest version of the EUPL.
-*  Please see LICENSE file for your rights under this license. */
-
-// Sanitize SERVER_NAME output
-$serverName = htmlspecialchars($_SERVER["SERVER_NAME"]);
-// Remove external ipv6 brackets if any
-$serverName = preg_replace('/^\[(.*)\]$/', '${1}', $serverName);
-
-// Set landing page location, found within /var/www/html/
-$landPage = "../landing.php";
-
-// Define array for hostnames to be accepted as self address for splash page
-$authorizedHosts = [ "localhost" ];
-if (!empty($_SERVER["FQDN"])) {
-    // If setenv.add-environment = ("fqdn" => "true") is configured in lighttpd,
-    // append $serverName to $authorizedHosts
-    array_push($authorizedHosts, $serverName);
-} else if (!empty($_SERVER["VIRTUAL_HOST"])) {
-    // Append virtual hostname to $authorizedHosts
-    array_push($authorizedHosts, $_SERVER["VIRTUAL_HOST"]);
-}
-
-// Determine block page type
-if ($serverName === "pi.hole"
-    || (!empty($_SERVER["VIRTUAL_HOST"]) && $serverName === $_SERVER["VIRTUAL_HOST"])) {
```

Comment:
@gstrauss I totally forgot to mention/ask - are we able to reproduce this in `pihole-admin.conf` where the following lines are? 

https://github.com/pi-hole/pi-hole/blob/ec1d4c550001b296cebd7bfac9faa7a8abaaf6df/advanced/pihole-admin.conf#L62-L66

If my docs reading is correct, is it something like the following?

```
 $HTTP["host"] == "pi.hole"  ||  $HTTP["host"] == env.VIRTUAL_HOST { 
     $HTTP["url"] == "/" { 
         url.redirect = ("" => "/admin/") 
     } 
 } 
```

Reply:
lighttpd.conf syntax is historically its own custom syntax (for better *and* worse), and is not shell syntax.  If VIRTUAL_HOST is set in the shell environment before starting lighttpd, then this syntax should work:
```
$HTTP["url"] == "/" {
    $HTTP["host"] == "pi.hole" {
         url.redirect = ("" => "/admin/")
    }
    $HTTP["host"] == env.VIRTUAL_HOST {
         url.redirect = ("" => "/admin/")
    }
} 
```
but the above will fail if VIRTUAL_HOST is not set.  It should at least be set to `VIRTUAL_HOST=""` to work
(Constructing a regex using lighttpd.conf syntax is possible but ugly and verbose
e.g. `^(" + env.VIRTUAL_HOST + "|pi.hole)$`)
That also requires VIRTUAL_HOST be set in the shell environment, and not missing from the shell env.

Another option, which I know is in wide use, is to use lighttpd.conf `include_shell` to run a short shell script which generates the simpler lighttpd.conf syntax to do what is desired, which could generate the syntax I posted above for a list of pi-hole admin hosts ("127.0.0.1", "::1", "localhost", "pi.hole", etc)

Lastly, a few lines of lua can be written and used with lighttpd mod_magnet.  Lua is a programming language and is much more expressive than lighttpd.conf syntax.  Whenever someone asks "can I do this arbitrarily complex thing in lighttpd.conf?", the answer frequently includes: *one option* is to use lighttpd mod_magnet and a lua script.

Maybe not as simple as you had hoped, but there are solutions available.  If you want to extend what is currently checked in, I would suggest `include_shell` and a shell script to generate the simpler lighttpd.conf directives to do exactly what you want.  I think requiring VIRTUAL_HOST be set in the environment will complicate things for various existing scripts in the OS which check `lighttpd -f /etc/lighttpd.conf -tt` and expect it to work (without knowing that VIRTUAL_HOST needs to be set in the environment)

Reply:
BTW, the "better" or advantages of lighttpd.conf syntax is realized in how lighttpd uses lighttpd.conf: lighttpd parses and generates a static configuration at startup, with limited conditionals allowed by the syntax.  Processing the lighttpd configuration for each and every request at runtime is much faster than runtime config parsing done by many other web servers.

Reply:
Intercepting and redirecting "/" might be desirable for new installations, but not for existing lighttpd configs which are used for other purposes.  If you do use `include_shell` and create a shell script for this configuration, instead of adding it to pihole-admin.conf, you might consider a new file pihole-admin-redir.conf so that some people can more easily disable just that part.

Alternatively -- and I think the best solution for this very-specific pi-hole use -- you can construct a simple index.html that you drop into the document root if `index.*` is not already present, and that index.html can be a splash page and/or can do the redirection in the HTML `<meta http-equiv="Refresh" content="0; url='/admin/'" />`

Reply:
Thanks for the information.

On reflection, I have decided that we do not need to include the functionality - But I spotted it and realised that I had overlooked it before, so wanted to find a way back in case we needed it.